### PR TITLE
[PM-3504] Return early for SAML auths

### PR DESCRIPTION
**Pull request by jlf0dev**
## Type of change

```
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Code changes

* **file.ext:** Description of what was changed and why

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: bitwarden_license/src/Sso/Utilities/Saml2BitHandler.cs [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.Extensions.Options;
+using Sustainsys.Saml2.AspNetCore2;
+using Sustainsys.Saml2.WebSso;
+
+namespace Bit.Sso.Utilities;
+
+public class Saml2BitHandler : IAuthenticationRequestHandler
```
Comment (withinfocus) [inline bitwarden_license/src/Sso/Utilities/Saml2BitHandler.cs:11]: 🎨 Maybe add a comment about where all this content came from -- I assume it's 99% copied.

Comment (jlf0dev) [reply bitwarden_license/src/Sso/Utilities/Saml2BitHandler.cs:11]: Will do, just wanted to validate this before cleaning it up 👍

Comment (jlf0dev) [reply bitwarden_license/src/Sso/Utilities/Saml2BitHandler.cs:14]: Comment added

**Review by withinfocus — COMMENTED** [review]
Looks good to me, but should get team review and testing.

**Review by JaredSnider-Bitwarden — APPROVED** [review]
LGTM!

**Pull request merged**